src/VectorizeLoops.cpp

```diff
@@ -994,6 +1240,169 @@ class VectorSubs : public IRMutator {
         : var(std::move(v)), replacement(std::move(r)), target(t), in_hexagon(in_hexagon) {
         widening_suffix = ".x" + std::to_string(replacement.type().lanes());
     }
+};  // namespace
+
+class FindVectorizableExprsInAtomicNode : public IRMutator {
+    Scope<> poisoned_names;
+    bool poison = false;
+
+    using IRMutator::visit;
+
+    template<typename T>
+    const T *visit_let(const T *op) {
+        mutate(op->value);
+        ScopedBinding<> bind_if(poison, poisoned_names, op->name);
+        mutate(op->body);
+        return op;
+    }
+
+    Stmt visit(const LetStmt *op) override {
+        return visit_let(op);
+    }
+
+    Expr visit(const Let *op) override {
+        return visit_let(op);
+    }
+
+    Expr visit(const Load *op) override {
+        // Even if the load is bad, maybe we can lift the index
+        IRMutator::visit(op);
+
+        // TODO: tuples?
```

Comment:
this TODO could probably use a bit more meat on the bone

Reply:
Hrm, that's something I genuinely forgot to finish. Will need to add a test.